### PR TITLE
Make MaxAndArgmax use internally signed axis numbers.

### PR DESCRIPTION
**Pull request by obilaniu**
This avoids an infinite loop when counting ndX downto 0 with unsigned
integers, as the condition `unsigned >= 0` always holds.

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins test this
add to whitelist

**Comment by nouiz** [comment]
if you can add a test that was failing and now pass, it would be great. I let @abergeron check in more detail the code.

**Comment by nouiz** [comment]
jenkins test this

**Comment by abergeron** [comment]
Nothing was failling before since there was a check on the arguments to prevent it.  However it made the reduce-all-axes case return an error.

File: src/gpuarray_buffer_cuda.c [hunk 1]
```diff
@@ -1305,6 +1305,10 @@ static int cuda_callkernel(gpukernel *k, unsigned int n,
     }
 
     switch (n) {
+    case 0:
+      ctx->err = cuLaunchKernel(k->k, 1, 1, 1, 1, 1, 1, shared,
+                                ctx->s, args, NULL);
+      break;
```
Comment (abergeron) [inline src/gpuarray_buffer_cuda.c]: I'm not sure we want to support 0-dim at this level.  This is basically useless.

If you really need it special case it where you do the kernel call.

Comment (obilaniu) [inline src/gpuarray_buffer_cuda.c]: @abergeron I amended the commit, that file is no longer touched. However, I do think it made sense. The pattern is pretty clear:

- 3-dims is a volume XYZ
- 2-dims is a matrix XY with Z=1
- 1-dims is a vector X with Y=Z=1
- 0-dims is a scalar with X=Y=Z=1

Comment (abergeron) [inline src/gpuarray_buffer_cuda.c]: The pattern is clear but running a 1, 1, 1 kernel is almost always a mistake and I don't want to encourage it.

**Pull request merged**